### ISSUE TITLE
Allow omitting parentheses for Expression.asA(Expression other)

### ISSUE THREAD

**Issue by Robbendebiene** [dart-lang/code_builder#436]
`Expression.asA(Expression other)` always wraps the code into parentheses. While this is a good default it is not always necessary. The problem for me is that the generated code triggers the linter rule [unnecessary_parenthesis](https://dart.dev/tools/linter-rules/unnecessary_parenthesis) in our CI. Unfortunately [dart_style auto fixes](https://pub.dev/documentation/dart_style/latest/dart_style/StyleFix-class.html) does not support the removal of unnecessary parenthesis.

My suggestion would be to have an optional parameter.
the following:

https://github.com/dart-lang/code_builder/blob/006790d8f3245124cc322bdf50376536ba28d85c/lib/src/specs/expression.dart#L68-L73

becomes

```dart
  Expression asA(Expression other, { bool parenthesized = true }) {
     final exp = BinaryExpression._( expression, other,  'as'));
     return parenthesized
        ? ParenthesizedExpression._(exp)
        : exp
  }
```

**Comment by natebosch** [comment]
We typically recommend avoiding that type of style lint for generated code. As mentioned in #213 readability is not a goal. This is a relatively easy change though, and most of the complexity stays in the calling code deciding what value to pass.

@srawlins do you have any opinions on adding this option?

**Comment by srawlins** [comment]
Yeah I think that supporting a `parenthesized` parameter everywhere (it's not just `asA` right?) would add a fair amount of complexity.

I recommend using `// ignore_for_file: type=lint` for generated files. See https://dart.dev/tools/analysis#suppressing-rules-for-a-file.
